### PR TITLE
Refractor Models to Freezed

### PR DESCRIPTION
**Pull request by Denzel101**
## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: mobile/pubspec.yaml
Comment (mboyamike) [inline mobile/pubspec.yaml:1]: When using riverpod, we won't really need get_it. All the service location can be done by riverpod

File: mobile/lib/utils/locator.dart
Comment (mboyamike) [inline mobile/lib/utils/locator.dart:1]: We'll use riverpod to inject the AuthRepository throughout the app. So I would also remove these lines

File: mobile/lib/auth/domain/repositories/auth_repository.dart
Comment (mboyamike) [inline mobile/lib/auth/domain/repositories/auth_repository.dart:1]: Looks good to me

File: mobile/lib/auth/data/models/user_model.dart
Comment (mboyamike) [inline mobile/lib/auth/data/models/user_model.dart:1]: Looks good to me

**Review by mboyamike — CHANGES_REQUESTED** [review]
We won't really need get_it since all the service location will be handled by riverpod. Removing the dependency should work fine

Also, just make the changes on the same branch and commit. I'll be able to see the changes in the pull request. No need to create a new pull request with the updated changes

**Review by mboyamike — APPROVED** [review]

**Comment by mboyamike** [comment]
Thanks for this

**Comment by Denzel-DM** [comment]
Don't mention it. You can close the issue now. If you need more help don't hesitate to reach out 🤙

**Pull request merged**